src/states/node.rs

```diff
@@ -2096,7 +2123,25 @@ impl Node {
                    self,
                    pub_id);
             let _ = self.crust_service.disconnect(pub_id);
-            let _ = self.peer_mgr.remove_peer(pub_id);
+            if let Some((peer, _)) = self.peer_mgr.remove_peer(pub_id) {
+                match *peer.state() {
+                    PeerState::Bootstrapper(_) |
+                    PeerState::Client => {
+                        let _ = self.dropped_clients.insert(*pub_id, ());
+                    }
+                    PeerState::ConnectionInfoPreparing { .. } |
+                    PeerState::ConnectionInfoReady(_) |
+                    PeerState::CrustConnecting |
+                    PeerState::Connected(_) |
+                    PeerState::SearchingForTunnel |
+                    PeerState::JoiningNode |
+                    PeerState::Routing(_) |
+                    PeerState::Candidate(_) |
+                    PeerState::Proxy => (),
+                }
+            } else {
+                let _ = self.dropped_clients.insert(*pub_id, ());
```

Comment:
why need to do this?
and does it mean the `dropped_clients` may contain non-client peer?